src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -110,14 +110,34 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
                 logging.warning(msg)
                 results.put({'name': name, 'success': "error", 'error': msg})
             else:
-                msg = "Error trying to process block %s through DBS. Error: %s" % (name, exString)
+                reason = parseDBSException(exString)
+                msg = "Error trying to process block %s through DBS. Error: %s" % (name, reason)
                 logging.exception(msg)
                 logging.debug("block info: %s \n", block)
                 results.put({'name': name, 'success': "error", 'error': msg})
 
     return
 
 
+def parseDBSException(exc):
+    """
+    parse DBS Go-based server exception
+    :param exc: exception message
+    :return: either (parsed) concise exception message or original exception
+    """
+    try:
+        data = json.loads(exc)
```

Comment:
are you sure the dbs3-client returns a json object here?

Reply:
the HTTP error body comes from dbs3-client which gets it from DBS server